src/cmd/services/m3coordinator/ingest/carbon/ingest_test.go

```diff
@@ -302,6 +334,158 @@ func TestIngesterHonorsPatterns(t *testing.T) {
 	}, found)
 }
 
+func TestIngesterNoStaticRules(t *testing.T) {
+	ctrl := xtest.NewController(t)
+	defer ctrl.Finish()
+
+	var expectationErr error
+	mockDownsamplerAndWriter, found := newMockDownsamplerAndWriter(ctrl, func(mappingRules []downsample.AutoMappingRule) {
+		// Use panics instead of require/assert because those don't behave properly when the assertion
+		// is run in a background goroutine.
```

Comment:
nit: better to set some persistent error than panic maybe?

Reply:
Ah, this comment is out of date since that's what we're doing now. Will remove.